### ISSUE TITLE
Error in DocumentFetcherStage if redirects for http connection are disabled

### ISSUE THREAD

**Issue by a-halubotski** [Norconex/collector-http#124]
Version: 2.1.0
There is an issue if the HttpRequest for document fetcher is configured with disabled redirects.

Once the redirect is disabled, the call `TargetURLRedirectStrategy.fixRedirectURL(...)` in method:
`HttpImporterPipeline$DocumentFetcherStage.executeStage(...)` is broken.

Because in this case the `TargetURLRedirectStrategy.CURRENT_URL` is never initialized.

How to reproduce:
Extend from `GenericDocumentFetcher`.
Override 
<pre><code>protected HttpRequestBase createUriRequest(final HttpDocument doc) {
    final HttpRequestBase request = new HttpGet(doc.getReference());
    request.setConfig(RequestConfig.custom().setRedirectsEnabled(false).build());
    return request;
}
</pre></code>

**Comment by essiembre** [comment]
This issue is now fixed in latest [2.2.0 snapshot release](http://www.norconex.com/collectors/collector-http/download).  

You may want to consider using the following crawler config snippet instead to disable redirect support, removing the need to extend any class:

```xml
<httpClientFactory>
  <maxRedirects>0</maxRedirects>
</httpClientFactory>
``` 
You do not need to specify the ``class`` when using the default implementation, which in this case is [GenericHttpClientFactory](http://www.norconex.com/collectors/collector-http/latest/apidocs/com/norconex/collector/http/client/impl/GenericHttpClientFactory.html).

**Comment by essiembre** [comment]
Norconex HTTP Collector 2.2.0 official release is out. It includes this fix. You can download it [here](http://www.norconex.com/collectors/collector-http/download).
